### PR TITLE
doc($app): revise document for developer guide

### PR DESCRIPTION
**Pull request by a0716116z**
Description :

1. Update getting start doc
2. Add Some pictures in google-drive-api.md
3. remove --without flag from bundle install command, since deprecated
4. fix the command error of rake spec
5. add descriptions to DB_KEY and MSG_KEY in secrets.yml
6. Add a picture for "OAuth consent screen" in google-drive-api.md step 2.
7. Fix type error from "groung" to "ground" in google-drive-api.md step 4.
8. Add three pictures for "selecting the scopes" in google-drive-api.md step 4.
9. Fix type error from "secret.yml" to "secrets.yml" in getting_start.md setup 3.
10. Add "rake" between RACK_ENV=test and db:migrate in getting_start.md setup 4.
11. Add a instruction about failing to run the app in getting_start.md setup 7.

### CODE REVIEW

**Review by Hsuan0923 — APPROVED** [review]

File: config/secrets_example.yml [hunk 1]
```diff
@@ -3,8 +3,8 @@ development:
   APP_URL: http://localhost:9090
   TZ: Asia/Taipei
   # Security
-  DB_KEY: 
-  MSG_KEY: 
+  DB_KEY: <32-bit base64 string>
```
Comment (soumyaray) [inline config/secrets_example.yml:6]: Do we have a rake task to generate these keys?  If so, perhaps we can say something like `<use rake db:key>` as the value string.

Comment (soumyaray) [reply config/secrets_example.yml:6]: and same for `MSG_KEY`

Comment (soumyaray) [reply config/secrets_example.yml:6]: oh I see – it's in `doc/getting_start.md` - might be worth mentioning here too?  up to you!

File: doc/getting_start.md [hunk 2]
```diff
@@ -20,22 +27,29 @@
...
 > please [install libsodium](https://github.com/RubyCrypto/rbnacl/wiki/Installing-libsodium) first
 
-3. Add config/secret.yml file
+3. Add config/secrets.yml file
 
 4. Run DB migrations
 ```
 $ rake db:migrate
-$ RACK_ENV=test db:migrate
+$ RACK_ENV=test rake db:migrate
```
Comment (soumyaray) [inline doc/getting_start.md:52]: I recommend using `RACK_ENV=test bundle exec rake db:migrate` to ensure the right gems are used.

File: doc/getting_start.md [hunk 3]
```diff
@@ -47,6 +61,20 @@ $ rake spec
 ```
 $ rake run:dev
 ```
+- If you fail to run the app, please try to install heroku installer first! (this command is for macOS)
+```
+$ brew tap heroku/brew && brew install heroku
+```
+- If you have the problem about redis after login to your account, plz run these commands:
```
Comment (soumyaray) [inline doc/getting_start.md:68]: please use 'please' instead of 'plz' :D

Comment (soumyaray) [reply doc/getting_start.md:68]: and elsewhere too

File: doc/google/google-drive-api.md [hunk 4]
```diff
@@ -5,29 +5,48 @@ It uses `google-api-client` package which is [deprecated](https://github.com/goo
 ### Step1: A GCP project with the API enabled
 
 1. [create GCP project](https://developers.google.com/workspace/guides/create-project#create_a_new_google_cloud_platform_gcp_project)
-2. [enable Google Drive API and Google Calendar API](https://developers.google.com/workspace/guides/create-project#enable-api)
-3. Button of "Enable APIs and Services" is hard to find 
-
+![](https://i.imgur.com/qiTpYxr.png) （Should accept the Service Rule when first time using This）
```
Comment (soumyaray) [inline doc/google/google-drive-api.md:8]: Should this png file be stored locally in `doc/images/` or somewhere like that?  I don't like the idea of images stored on someone's imgur account or elsewhere.

Comment (soumyaray) [reply doc/google/google-drive-api.md:8]: same with all the other imgur links below.  that way we can also name the files something useful.

File: doc/test_code.md [hunk 5]
```diff
@@ -1,7 +1,7 @@
 # Run the tests
 1. Migrate test.db
 ```
-RACK_ENV = test rake db:migrate
+RACK_ENV=test rake db:migrate
```
Comment (soumyaray) [inline doc/test_code.md:4]: again, I'd suggest: `RACK_ENV=test bundle exec rake db:migrate`

**Review by soumyaray — CHANGES_REQUESTED** [review]
Hope my review comments help! Let me know if you have any questions!

File: config/secrets_example.yml (hunk 1 above)
Comment (a0716116z) [reply config/secrets_example.yml:6]: ok ~ I will change this part

File: doc/google/google-drive-api.md (hunk 4 above)
Comment (a0716116z) [reply doc/google/google-drive-api.md:8]: I think we could change this in next branch

Comment (soumyaray) [reply doc/google/google-drive-api.md:8]: agreed and that's a good idea! we can do that along with the new architecture diagrams.

**Review by soumyaray — CHANGES_REQUESTED** [review]

**Review by soumyaray — APPROVED** [review]
Thanks for the great work!

**Pull request merged**